### PR TITLE
5815 signup specific domains

### LINKED ISSUE #5815: [3.0 feature request] Signup only with specific Email domains
Would it be possible to enable signing up only with specific Email domains when an external authentication is used such as microsoft365)?

It was possible on v2 (#1548), but on v3 I do not find how to set it up.
Maybe Keycloak has such function?

(I thought I posted this request somewhere, but I cannot find it.)

### PR DESCRIPTION
**Pull request by alihadimazeh**
resolves #5815

### CODE REVIEW

File: app/assets/locales/en.json [hunk 1]
```diff
@@ -319,7 +319,10 @@
           "open": "Open Registration",
           "invite": "Join by Invitation",
           "approval": "Approve/Decline"
-        }
+        },
+        "specific_email_domain_signup": "Specific Email Domain Signup",
+        "specific_email_domain_signup_description": "Allow specific email domains to sign up. Format must be: @test.com,domain.com",
```
Comment (farhatahmad) [inline app/assets/locales/en.json:324]: "Allow `only` specific ...."

File: Gemfile.lock [hunk 2]
```diff
@@ -551,3 +551,9 @@ DEPENDENCIES
   web-console (>= 4.2.1)
   webdrivers
   webmock
+
+RUBY VERSION
+   ruby 3.1.0p0
+
+BUNDLED WITH
+   2.3.3
```
Comment (farhatahmad) [inline Gemfile.lock:559]: This should be removed

File: app/assets/locales/en.json [hunk 3]
```diff
@@ -319,7 +319,10 @@
           "open": "Open Registration",
           "invite": "Join by Invitation",
           "approval": "Approve/Decline"
-        }
+        },
+        "allowed_domains": "Allow Specific Email Domains",
```
Comment (farhatahmad) [inline app/assets/locales/en.json:323]: I would just put this as "Allowed Email Domains"

File: app/assets/locales/en.json [hunk 4]
```diff
@@ -420,7 +423,8 @@
         "privacy_policy_updated": "The privacy notice has been updated.",
         "helpcenter_updated": "The help center link has been updated.",
         "terms_of_service_updated": "The terms of service have been updated.",
-        "maintenance_updated": "The maintenance banner has been updated."
+        "maintenance_updated": "The maintenance banner has been updated.",
+        "allowed_domains_signup_updated": "The specific email domain sign up has been updated"
```
Comment (farhatahmad) [inline app/assets/locales/en.json:427]: "The allowed email domains has been updated"

File: db/data/20240806205559_add_domain_specific_email_signup_to_site_settings.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  def down
+    Tenant.find_each do |tenant|
+      SiteSetting.find_by(setting: Setting.find_by(name: 'Maintenance'), provider: tenant.name)&.destroy
+    end
+
+    SiteSetting.find_by(setting: Setting.find_by(name: 'Maintenance'), provider: 'greenlight')&.destroy
+
+    Setting.find_by(name: 'SpecificEmailDomainSignUp')&.destroy
+  end
+end
```
Comment (farhatahmad) [inline db/data/20240806205559_add_domain_specific_email_signup_to_site_settings.rb:23]: This can be removed

File: db/schema.rb [hunk 6]
```diff
@@ -193,12 +193,12 @@
...
     t.boolean "verified", default: false
     t.string "verification_digest"
-    t.datetime "verification_sent_at", precision: nil
+    t.datetime "verification_sent_at"
     t.string "session_token"
-    t.datetime "session_expiry", precision: nil
+    t.datetime "session_expiry"
     t.integer "status", default: 0
     t.index ["email", "provider"], name: "index_users_on_email_and_provider", unique: true
     t.index ["reset_digest"], name: "index_users_on_reset_digest", unique: true
```
Comment (farhatahmad) [inline db/schema.rb:204]: This shouldn't be included

Comment (alihadimazeh) [reply db/schema.rb:204]: the change should not be included, or the file itself?

**Pull request closed without merging**